### PR TITLE
feat(auth): refactor route mounting to be more easily readable

### PR DESCRIPTION
**Pull request by njlie**
## Changes proposed in this pull request

- Clearer enumeration of all the routes that are being mounted to the app

## Context

The old pattern of iterating through a list of routes and mounting each item as a route was not great for readability, so they are individually mounted so the different paths and methods can be seen in the code.

## Checklist

- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: packages/auth/src/app.ts [hunk 1]
```diff
@@ -189,83 +189,137 @@ export class App {
...
+      createValidatorMiddleware<
+        ContextType<(ctx: AppContext) => Promise<void>>
+      >(openApi, {
+        path: '/interact/{id}/{nonce}/finish',
+        method: HttpMethod.GET
+      }),
+      grantRoutes.interaction.finish
+    )
+
+    // Grant lookup
```
[the remark this answers is not in the document]
Comment (njlie) [reply packages/auth/src/app.ts:288]: We could adopt the same behavior we do with the `/introspect` endpoint and have the authentication be optional. If the option to authenticate is set, the IDP could pre-register in the same manner as an RS to associate its public key with some identifier and use that. That probably falls into the scope of https://github.com/interledger/rafiki/issues/600

File: packages/auth/src/app.ts [hunk 2]
```diff
@@ -189,83 +189,117 @@ export class App {
...
-                ctx.status = 200
-              }
-            )
-          }
-        }
-      }
-    }
+    // Grant Continue
+    this.publicRouter.post(
+      '/continue/id',
```
Comment (wilsonianb) [inline packages/auth/src/app.ts:206]:
```suggestion
      '/continue/:id',
```

File: packages/auth/src/app.ts [hunk 3]
```diff
@@ -189,83 +189,117 @@ export class App {
...
+      createValidatorMiddleware(openApi, {
+        path: '/continue/{id}',
+        method: HttpMethod.POST
+      }),
+      signatureService.tokenHttpsigMiddleware,
+      grantRoutes.continue
+    )
```
Comment (wilsonianb) [inline packages/auth/src/app.ts:213]: The configured (not latest) spec also has `PATCH` and `DELETE` for `/continue/{id}`.
https://github.com/interledger/open-payments/blob/53236cc92a070ec98a7ab31c6ed0f6fc50a98041/auth-server-open-api-spec.yaml#L200-L299
I'm surprised our old magic `for` loop(s) didn't complain :grimacing:

Comment (njlie) [reply packages/auth/src/app.ts:213]: Interesting...will update to use the latest spec.

**Review by wilsonianb — APPROVED** [review]

**Pull request merged**